keps/sig-security/NNNN-ambient-capabilities/README.md

```diff
@@ -0,0 +1,889 @@
+and make progress.
+-->
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+#### Story 2
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Changes to kubernetes API (https://pkg.go.dev/k8s.io/api/core/v1)
+
+Reusing the existing capabilities field in the securityContext might cause confusion and we propose adding a new field in the SecurityContext called ambientCapabilities. 
+
+
+```go
+type SecurityContext struct {
+	// The capabilities to add/drop when running containers.
+	// Defaults to the default set of capabilities granted by the container runtime.
+	// +optional
+	Capabilities *Capabilities `json:"capabilities,omitempty" protobuf:"bytes,1,opt,name=capabilities"`
+     // The ambient capabilities to add when running containers as non-root.
+     // +optional
+     AmbientCapabilities []Capability `json:"ambientCapabilities,omitempty" protobuf:"bytes,1,rep,name=ambientCapabilities,casttype=[]Capability"`
+
+// Rest of the struct omitted for brevity. You can find it at https://pkg.go.dev/k8s.io/api/core/v1#SecurityContext 
+```
+
+
+Notes about how ambientCapabilities would work:
+
+1. AmbientCapabilities can only be added.
+2. If the field is not supplied kubernetes will not add any capabilities to the ambient capabilities set by default.
+3. AmbientCapabilities field should only be set if the Pod is running as non-root. There are 2 ways to handle this:
```

Comment:
Why? Maybe the container starts as root, but then drops privileges after initialization?

Reply:
Good catch! I did not think of this scenario, removed this constraint.